### PR TITLE
#39: Add Popover from labonline (semantic free!) (closes #39)

### LINKED ISSUE #39: Add Popover from labonline (semantic free!)

### PR DESCRIPTION
**Pull request by FrancescoCioria**
Issue #39

### CODE REVIEW

File: src/popover/Popover.js [hunk 1]
```diff
@@ -1,99 +1,308 @@
...
+        type: 'relative',
+        position: 'top',
+        anchor: 'center',
+        event: 'hover',
+        onShow: () => {},
+        onHide: () => {},
+        dismissOnScroll: true,
+        className: ''
+      },
+      ...this.props.popover
```
Comment (francescogior) [inline src/popover/Popover.js]: `{ ...{ prop1: value1, prop2: value2 }, ...rest }` should be the same of `{ prop1: value1, prop2: value2 , ...rest }`

File: src/popover/Popover.js [hunk 2]
```diff
@@ -1,99 +1,308 @@
...
-    const settings = {
-      on: 'manual',
-      popup: $(this.refs.popover.getDOMNode()),
-      onVisible: this.onVisible,
-      onHidden: this.onHidden,
-      hideOnScroll: false // TODO
+    this.saveValuesFromNodeTree();
+  },
+
+  saveValuesFromNodeTree(cb) {
```
Comment (francescogior) [inline src/popover/Popover.js]: Here and for all the methods shouldn't be safer to use fat arrows?

**Comment by francescogior** [comment]
:+1:

File: src/popover/Popover.js (hunk 2 above)
Comment (FrancescoCioria) [inline src/popover/Popover.js]: Nope as in react components we're not using ES6 classes

File: src/popover/Popover.js (hunk 1 above)
Comment (FrancescoCioria) [inline src/popover/Popover.js]: :+1:

**Comment by FrancescoCioria** [comment]
@giogonzo maybe you want to give it a look too as we'll be using it in LOL soon

**Comment by giogonzo** [comment]
looks good to me

- can you summarize additions with respect to LOL popover? is it just handling both absolute and relative to DOM?
- looks safe for iso rendering, but could you try it once? It should be as easy as:
```js
document.getElementById('container').innerHTML = React.renderToString(template);
// instead of:
// React.render(template, document.getElementById('container'));
```
(actually this could be a nice test for each component, maybe rendered side by side in examples?)

**Comment by FrancescoCioria** [comment]
I also changed the API: I grouped all the props related to popover inside an object
see here: https://github.com/buildo/react-components/commit/861b6cf69c9324982b0b43ec91f5b7ee4d50185a

I added a global isomorphic test as suggested. It works fine

**Comment by giogonzo** [comment]
nice
